### DISCUSSION TITLE
WordPress Plugin

### DISCUSSION THREAD

**Discussion by nirjhar** [immich-app/immich#6438] [General]
I am a photographer. I am using immich to manage my library. I'd love to have a wordpress plugin of immich from where I can add my photos in my blog using regular expression search from my immich library. Also, I'd love to have custom tag field in immich meta data. Can you guys follow the flickr? 

Cheers.

**Comment by bo0tzz** [comment]
I doubt this is something the Immich team will build, but it should be possible for anybody else to do so on top of our API.

**Comment by defcon1702** [comment]
I started to build one. Works pretty well till now (Albumsync, Happy Files Support, CPT creation). Will need some more work but I will come back when I have a testable version. :)

**Reply by nirjhar** [reply]
Please share it so that we can give you feedback :)

**Reply by mstewart14** [reply]
@defcon1702 I'd be interested in helping with code if you're open to that...

**Comment by shaneoreilly** [comment]
@defcon1702 I'd love to help with testing this, once it's ready

**Comment by palitu** [comment]
@defcon1702  - that would be fantastic if you could share it.

I am basically after a way of embedding some photo galleries into a post, so that people can click through.  Then i can add the descriptions to the photo in immich, and it can display in WP.

cheers

**Comment by palitu** [comment]
in case anyone was interested, i have put something together that works for me.  it basically takes an Immich album, and uploads all the photos in it to WP media.  it also assigns the immich photo description to the WP photo caption.

I would like it to auto generate the post with a gallery, but i have some authentication issues there.  

Feel free to use, modify, distribute as you please.  I am also happy to take PR's, but please do not comment in its quality... i already know :( 

hope it helps someone!

https://github.com/palitu/immichSync

**Comment by vogon1** [comment]
In case anyone is interested, I created a plugin that keeps the assets in Immich, and displays them on Wordpress. It is directly installable from Wordpress plugins.

https://nl.wordpress.org/plugins/gallery-for-immich/ or
https://github.com/vogon1/immich-wordpress-plugin
